src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -295,24 +298,41 @@ public void schedulePeriodicFleetStatusDataUpdate(boolean isDuringConnectionResu
                 initialDelay, periodicPublishIntervalSec, TimeUnit.SECONDS);
     }
 
+    private AtomicReference<OverallStatus> getOverallStatus() {
+        Instant now = Instant.now();
+        AtomicReference<OverallStatus> overallStatus = new AtomicReference<>(OverallStatus.HEALTHY);
+        this.kernel.orderedDependencies().forEach(service -> {
+            serviceFssTracksMap.put(service, now);
+            overallStatus.set(getOverallStatusBasedOnServiceState(overallStatus.get(), service));
+        });
+        return overallStatus;
+    }
+
```

Comment:
There are other places calling `getOverallStatusBasedOnServiceState` and looks like you refactor those as well